R/MzTabParam.R

```diff
@@ -0,0 +1,122 @@
+#' @title Store contents of an `XcmsExperiment` objects as a .mztabm file
+#'
+#' @name MzTabParam
+#'
+#' @export
+#'
+#' @inheritParams saveMsObject
+#'
+#' @family MS object export and import formats.
+#'
+#' @description
+#' The `saveMsObject()` and `readMsObject()` methods with the `MzTabParam`
+#' option enable users to save/load `XcmsExperiment` objects in/from a .mztabm
+#' file format. Mainly the metadata (MTD) and Small molecule feature (SMF)
+#' tables will represent the `XcmsExperiment`. The small molecule summary
+#' section (SML) will be filled with `null` values as no annotation and
+#' identification of compound is performed in `xcms`. Writing data to a folder
+#' that contains already exported data will result in an error.
+#'
+#' @param studyId `character(1)` Will be both the `filename` of the object
+#' saved in .mztabm format and the `mzTab-ID` in the file.
+#'
+#' @param polarity `character(1)` Describes the polarity of the experiment. Two
+#' inputs are possible, "positive" (default) or "negative".
+#'
+#' @param sampleDataColumn `character` strings corresponding to the column
+#' name(s) of the `sampleData()` of the `XcmsExperiment` object with the
+#' different *variables* of the experiment, for example it could be
+#' *"phenotype"*, *"sample_type"*, etc...
+#'
+#' @param path `character(1)` Define where the file is going to be stored. The
+#' default will be `tempdir()`.
+#'
+#' @param optionalFeatureColumns Optional columns from `featureDefinitions`
+#' that should be exported too. For example it could be *"peakidx*,
```

Comment:
Uh - do we really support exporting `"peakidx"`? just asking because that's a list of indices - and for large data sets it's might be (if pasted) a veeery long character. I would maybe, instead of listing these two examples, say that any column names of `featureDefinitions()` can be specified.

Reply:
Ye that's the worse example i could have given actually...